DOC: Improved doc for `Index.equals`

ShaharNaveh opened the pull request:
- [x] closes #33285
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Commit:
DOC: Improved doc for `Index.equals`

mroeschke commented on pandas/core/indexes/base.py:
```diff
@@ -4193,15 +4193,43 @@ def putmask(self, mask, value):
+        >>> idx1.equals(pd.Index([1, 2, 3]))
+        True
+
+        The dtype is been compared as well
```
`is been` -> `is`

jorisvandenbossche replied:
The dtype is actually *not* compared:

```
In [9]: pd.Int64Index([1, 2, 3]).equals(pd.UInt64Index([1, 2, 3]))                                                                                                                            
Out[9]: True
```

(in your example here, the values are not equal as a string is not equal to an integer)

mroeschke commented on pandas/core/indexes/base.py:
```diff
@@ -4193,15 +4193,43 @@ def putmask(self, mask, value):
+        >>> idx1.equals(idx2)
+        False
+
+        The oreder is also been compared
```
`oreder is also been compared` -> `order is also compared`

jorisvandenbossche commented on pandas/core/indexes/base.py:
```diff
@@ -4193,15 +4193,43 @@ def putmask(self, mask, value):
+    def equals(self, other: Any) -> bool:
         """
-        Determine if two Index objects contain the same elements.
+        Determine if two Index objects contain the same elements with the same order.
```
Or we could also simplify "... contain the same elements with the same order" to "... are equal" ?

We can in a second paragraph detail on what is considered as equal

jorisvandenbossche reviewed (commented):
While you are at it, can you also add a "Parameters" section?

Commit:
Merge remote-tracking branch 'upstream/master' into DOC-Imporve-IndexEquals

Commit:
Grammer

XREF:
https://github.com/pandas-dev/pandas/pull/33289#discussion_r403509327
https://github.com/pandas-dev/pandas/pull/33289#discussion_r403509363

Commit:
Added parameters sections, and a second peragraph

XREF:
https://github.com/pandas-dev/pandas/pull/33289#pullrequestreview-387790007
https://github.com/pandas-dev/pandas/pull/33289#discussion_r403662373

Commit:
Fix examples

jorisvandenbossche commented on pandas/core/indexes/base.py:
```diff
@@ -4195,7 +4195,17 @@ def putmask(self, mask, value):
+
+        The things that are being compared are:
+
+            * The elements inside the Index object.
+            * The order of the elements inside the Index object.
```
```suggestion
        * The elements inside the Index object.
        * The order of the elements inside the Index object.
```

ShaharNaveh committed:
Update pandas/core/indexes/base.py

Co-Authored-By: Joris Van den Bossche <jorisvandenbossche@gmail.com>

jorisvandenbossche commented:
Thanks!
